ceph-salt-formula/salt/ceph-salt/ceph-admin.sls

```diff
@@ -0,0 +1,20 @@
+{% import 'macros.yml' as macros %}
+
+{% if pillar['ceph-salt']['minions']['admin'] | length > 0 %}
+
+{{ macros.begin_stage('Deployment of Admin nodes') }}
+
+copy ceph.conf and keyring to other admin nodes:
+  cmd.run:
+    - name: |
+{%- for minion in pillar['ceph-salt']['minions']['admin'] %}
+{%- if minion != grains['host'] %}
+        scp -o "StrictHostKeyChecking=no" /etc/ceph/ceph.conf root@{{ minion }}:/etc/ceph/
+        scp -o "StrictHostKeyChecking=no" /etc/ceph/ceph.client.admin.keyring root@{{ minion }}:/etc/ceph/
+{%- endif %}
+{%- endfor %}
+    - failhard: True
+
+{{ macros.end_stage('Deployment of Admin nodes') }}
```

Comment:
Here I'm concerned what will happen if there is no "separate" Admin node to copy files to. It seems like ceph-salt will produce output like this:

```
[STEP] [BEGIN] Deployment of Admin nodes
[STEP] [END] Deployment of Admin nodes
```

Won't the user find this confusing if he did not define any Admin role?

Also, what will happen when Salt runs the `cmd.run` state with an empty `name`?

In other places I saw this code pattern:

```
{%- for minion in pillar['ceph-salt']['minions']['admin'] %}
{%- if minion != grains['host'] %}

copy ceph.conf and keyring to Admin node {{ minion }}:
  cmd.run:
    - name: |
        scp -o "StrictHostKeyChecking=no" /etc/ceph/ceph.conf root@{{ minion }}:/etc/ceph/
        scp -o "StrictHostKeyChecking=no" /etc/ceph/ceph.client.admin.keyring root@{{ minion }}:/etc/ceph/
    - failhard: True
{%- endif %}
{%- endfor %}
```

It seems like this would at least eliminate the empty `cmd.run` invocation.

Reply:
Should be fixed now, please review again.